implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/OtelMicroMeterSpanContextSupplier.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.prometheus;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanId;
+import io.opentelemetry.api.trace.TraceId;
+import io.prometheus.client.exemplars.tracer.common.SpanContextSupplier;
+import io.prometheus.client.exemplars.tracer.otel.OpenTelemetrySpanContextSupplier;
```

Comment:
Why would we need this?
We should not depend on OTel-specific things, we only want to depend on Prometheus' `SpanContextSupplier` so that users can setup the implementation of their choice (Otel, Brave, Sleuth, etc.).

Reply:
Hello @jonatan-ivanov.
I introduce this because prometheus java client also use this implementation too to give use spancontextsupplier "real" implementation (here : https://github.com/prometheus/client_java/blob/master/simpleclient_tracer/simpleclient_tracer_otel/src/main/java/io/prometheus/client/exemplars/tracer/otel/OpenTelemetrySpanContextSupplier.java) 

What is not done in prometheus client is to take account of Sampling and that's what i add in this above implementation of SpanContextSupplier. The goal is not use trace id that will be not recorded to only keep useful trace on backend (isSample take account of TraceFlags to indicate above service that this trace won't be recorded).

When we use Prometheus's SpanContextSupplier, transitively we use io.opentelemetry package too in fact. So i don't know how sleuth (that use brave) work but it's mandatory to these to also have this transitive dependency to open telemetry packages.

I'm agree that is prometheus client integrate "issamppled" in their code, this dependence would be less explicit. But it exists anyway.

Reply:
I also take a look at current Brave (so sleuth) implementation to be sure.

Actually, Brave depends on OpenTracing API (https://github.com/openzipkin/brave/blob/master/brave/pom.xml) but OpenTracing is also deprecated  in benefice of OpenTelemetry (https://github.com/opentracing/specification/issues/163)

These APIs (OpenTracing) will be archived at the end of this year. Project Brave (and indeed Sleuth) will have to remove this dependency.

Reply:
I create an issue on prometheus client for this case : I create an issue on prometheus client for this case : https://github.com/prometheus/client_java/issues/765 (it is like russian doll to get this feature :-))

Reply:
>I introduce this because prometheus java client also use this implementation too to give use spancontextsupplier "real" implementation
I think implementations should stay in their respective projects: OTel's `SpanContextSupplier` should be in OTel, Brave's `SpanContextSupplier` should be in Brave, Sleuth's `SpanContextSupplier` should be in Sleuth and so on. We can't have `SpanContextSupplier` implementations for every single tracing library.

>What is not done in prometheus client is to take account of Sampling
I guess if there is an issue with  Prometheus's SpanContextSupplier, that should be fixed in Prometheus not in Micrometer.

>When we use Prometheus's SpanContextSupplier, transitively we use io.opentelemetry package too in fact.
I don't think so: `io.prometheus.client.exemplars.tracer.common.SpanContextSupplier`. You can take a look at my draft for Counter exemplars, it does not contain OTel.

>So i don't know how sleuth (that use brave) work but it's mandatory to these to also have this transitive dependency to open telemetry packages.
I think we have a huge gap here. I don't see why OTel would be mandatory. What if I'm (as a user) not using it at all?

>Actually, Brave depends on OpenTracing API
The scope of that dependency is `test` so Brave uses it for testing something. :)

>These APIs (OpenTracing) will be archived at the end of this year. Project Brave (and indeed Sleuth) will have to remove this dependency.
Brave uses it for testing, Sleuth does not depend on it.

Reply:
That's ok @jonatan-ivanov .
I have a PR in progress on prometheus project to remove this dependency.
I take you inform.

Reply:
Hello @jonatan-ivanov.
Yes with this PR, nothing to do in micrometer (just some things to remove in my current PR for micrometer, like removing explicit dependencies to opentelemetry as you ask for).
But without it, we'll get "incorrect" TraceId in exemplars if sampling is applied (and i think everyone sample for traces).

Reply:
I'm not sur to clearly understand your question. I think yes, we must return null from SpanContextSupplier of prometheus client in this case. It is the goal of the PR i submit to java prometheus project.

Reply:
I mean if the `SpanContextSupplier` does this: [see SleuthSpanContextSupplier](https://github.com/spring-cloud/spring-cloud-sleuth/blob/3.1.x/spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/prometheus/prometheus/SleuthSpanContextSupplier.java), we don't need to wait till Prometheus merges your changes and releases a new version, we can go ahead with the Micrometer support as is.

Reply:
Yes it should work too. If you want i can test it next week. But we have to keep OtelMicroMeterSpanContextSupplier class (without sample method) and dependencies to otel api in this class. 
Are you agree with that ? (it seems to me that you don't want to have dependencies on otel api at all, it's why i created PR on prometheus java client side)